public/apps/configuration/panels/role-edit/role-edit.tsx

```diff
@@ -110,14 +143,39 @@ export function RoleEdit(props: RoleEditDeps) {
       try {
         setTenantNames(await fetchTenantNameList(props.coreStart.http));
       } catch (e) {
-        // TODO: show user friendly error message
-        console.log(e);
+        addToast({
+          id: 'tenant',
+          color: 'danger',
+          title: 'Failed to load data',
+          text:
+            'Failed to load data. You may refresh the page to retry or see browser console for more information.',
+        });
+        console.error(e);
       }
     };
 
     fetchTenantNames();
   }, [props.coreStart.http]);
 
+  const updateRoleHandler = async () => {
+    try {
+      await updateRole(props.coreStart.http, roleName, {
+        cluster_permissions: roleClusterPermission.map(comboBoxOptionToString),
+        index_permissions: unbuildIndexPermissionState(roleIndexPermission),
+        tenant_permissions: unbuildTenantPermissionState(roleTenantPermission),
+      });
+      // Redirect to role detail page
+    } catch (e) {
+      addToast({
+        id: 'updateRole',
+        color: 'danger',
+        title: `Failed to ${props.action} role`,
+        text: `Failed to ${props.action} role. You may refresh the page and then retry or see browser console for more information.`,
+      });
+      console.error(e);
```

Comment:
is this for testing, or we really want to log the error information in browser console?

Reply:
Not for testing. I expect the error trace could be long some I didn't add it to the toast. And don't really know what errors could happen. Any suggestion?